console/src/components/Services/Data/Add/ForeignKeyWrapper.js

```diff
@@ -105,6 +106,10 @@ const ForeignKeyWrapper = ({
 
     const expandCallback = () => {
       dispatch(toggleFk(i));
+      const newFks = JSON.parse(JSON.stringify(foreignKeys));
+      newFks[i].refSchemaName = event.target.value;
+      dispatch(setForeignKeys(newFks));
```

Comment:
I feel we shouldn't mutate `newFks` directly. Here's a suggestion to change that, if required that is:

```suggestion
      const newFks = JSON.parse(JSON.stringify(foreignKeys));
       const updatedFks = {
          ...newFks,
          [i]: {
              ...newFks[i],
              refSchemaName: event.target.value,
          },
      };
      dispatch(setForeignKeys(updatedFks));
```